### PR TITLE
feat(window): mocking window object

### PR DESCRIPTION
**Pull request by manekinekko**
PR for https://github.com/angular/universal/issues/282

### CODE REVIEW

File: modules/universal/server/src/node/mock/window.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  "onsuspend",
+  "ontimeupdate",
+  "onunload",
+  "onvolumechange",
+  "onwaiting"
+].map(beDefensive);
+
+export var window = win;
+global.window = win;
+GLOBAL.window = win;
```
Comment (gdi2290) [inline modules/universal/server/src/node/mock/window.ts]: what's `GLOBAL`?

Comment (manekinekko) [inline modules/universal/server/src/node/mock/window.ts]: in node, GLOBAL and global are equivalent
image

Comment (manekinekko) [inline modules/universal/server/src/node/mock/window.ts]: I set those two justs in case :)

**Comment by gdi2290** [comment]
kk merging

File: modules/universal/server/src/node/mock/window.ts (hunk 1 above)
Comment (gdi2290) [inline modules/universal/server/src/node/mock/window.ts]: it looks like `GLOBAL` and `root` will be removed
https://github.com/nodejs/node/pull/1838

**Comment by thebarada** [comment]
@gdi2290 how it works?
because I'm still getting error
`ReferenceError: window is not defined`

**Comment by thebarada** [comment]
@gdi2290

**Comment by manekinekko** [comment]
@thebarada please provide us with a working sample so we can reproduce your issue?

**Comment by gdi2290** [comment]
@thebarada  if you're talking about Material error there's a workaround in the universal-starter. you can track this issue for window "support"
https://github.com/angular/universal/issues/534

**Comment by Xaiid** [comment]
@gdi2290 I imported Ng1toNg2(Need Ng1toNg2 to make hybrid application routes works) module from ui-router, this requires angular 1.5.8 which uses window, and is throwing the error.

node_modules/angular/angular.js:31766
})(window);
   ^

ReferenceError: window is not defined`

Is Angular 1.5.8 going to be supported?

**Comment by gdi2290** [comment]
Angular 1.x.x is not supported for universal

**Comment by rsach** [comment]
i'm trying to use masonry.js in universal starter pack with a 2.3 config 
window isn't defined , is there any workaround sample that i can refer to 
thanks in advance

/node_modules/masonry-layout/masonry.js:33
}( window, function factory( Outlayer, getSize ) {
   ^

ReferenceError: window is not defined
    at Object.<anonymous> (

**Comment by sushrest** [comment]
Looks like its merged.. How to implement global window object for SSR. I am still getting undefined window error.

**Pull request merged**